### PR TITLE
refactor(blocks): theme observer singleton

### PR DESCRIPTION
**Pull request by fundon**
Closes: [BS-975](https://linear.app/affine-design/issue/BS-975/重构-theme-observer)

Use the theme observer singleton to monitor system theme switching.

* use `signals` instead of `slot`
* add helper functions
  * `getColorValue`
  * `getPropertyValue`
  * `generatorColorProperty`
* remove redundant functions and types
  *  `getThemeMode` 
  * `extractCssVariables` 
  * `CssVariableName`

### CODE REVIEW

**Comment by fundon** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7736?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#7736** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7736?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7721** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7721?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#7727](https://github.com/toeverything/blocksuite/pull/7727) <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7727?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#7585** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7585?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#7729](https://github.com/toeverything/blocksuite/pull/7729) <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7729?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/blocks/src/_common/theme/theme-observer.ts [hunk 1]
```diff
@@ -1,45 +1,32 @@
...
-export class ThemeObserver extends Slot<CssVariablesMap> {
-  private _cssVariables: CssVariablesMap | null = null;
+export class ThemeObserver extends Slot<ColorScheme> {
+  static #computedStyle: CSSStyleDeclaration;
 
-  private _mode: ColorScheme = ColorScheme.Normal;
+  static #instance: ThemeObserver;
 
-  private _observer?: MutationObserver;
+  #mode: ColorScheme = ColorScheme.Light;
```
Comment (Saul-Mirone) [inline packages/blocks/src/_common/theme/theme-observer.ts:18]: What if just make `mode` a signal?

Comment (fundon) [reply packages/blocks/src/_common/theme/theme-observer.ts:18]: I've thought about this too, it should be fine.

**Review by Saul-Mirone — APPROVED** [review]
LGTM

**Pull request merged**